### DISCUSSION TITLE
Let the user change the date/time

### DISCUSSION THREAD

**Discussion by JokerGermany** [emsesp/EMS-ESP32#1091] [Ideas]
My ems-esp got unresponsive.
Because i don't know how to restart it safe, I use the emergency button to shut of the power supply of the heating and put it on again. 
Sometimes the RC30 will lose the date/time and i have to configure it again.

It's a pitty, that this isn't possible with ems-esp.

**Comment by proddy** [comment]
You can set the time using EMS-ESP on an RC35 and an RC300. And that's only because we own the thermostat or someone from the community was willing to figure out the telegrams to use. If you want to help a good start is to monitor the incoming telegrams via the EMS-ESP log (or SysLog) while you manually set the time on the RC30.

And the RC30 should have a battery and an RTC (clock) so should never loose it's time, even if the power is off. Unless the battery is dead.

**Reply by JokerGermany** [reply]
The interesting thing is, that it only lose the time, nothing else...

**Reply by JokerGermany** [reply]
Will try to get the log next time.

**Reply by MichaelDvP** [reply]
RC30 and RC35 are mostly similar, but one of the differences is, that RC30 clock can't be set over ems-bus. For RC35 ems-esp checks the clock setting and updates from NTP if RC35 clock is wrong, but on RC30 that's not possible, sorry.

The clock is buffered by a goldcap, holding a few hours (older buderus use non recharchable Li-battery). The other settings are stored in an eeprom, but writing to eeprom (~1mio writecycles) every minute will kill the cells within a few years, so the clock uses battery buffered ram.
